modules/renter/files.go

```diff
@@ -107,6 +108,34 @@ func (f *file) uploadProgress() float64 {
 	return 100 * (float64(uploaded) / float64(desired))
 }
 
+// redundancy returns the redundancy of the least redundant chunk. A file
+// becomes available when this redundancy is >= 1. Assumes that every piece is
+// unique within a file contract. -1 is returned if the file has size 0.
+func (f *file) redundancy() float64 {
+	if f.size == 0 {
+		return -1
+	}
+	piecesPerChunk := make([]int, f.numChunks())
+	// We cannot get the redundancy if piecesPerChunk has 0 length because later
+	// in the function we use the first element of the slice.
+	if len(piecesPerChunk) == 0 {
+		build.Critical("cannot get redundancy of a file with 0 chunks")
+		return 0
+	}
+	for _, fc := range f.contracts {
+		for _, p := range fc.Pieces {
+			piecesPerChunk[p.Chunk]++
+		}
+	}
+	minPieces := piecesPerChunk[0]
+	for _, numPieces := range piecesPerChunk {
+		if numPieces < minPieces {
+			minPieces = numPieces
+		}
+	}
```

Comment:
caching is trivial to add later. We could add a `cachedRedundancy` field to the file object and update it after each upload.